core/types/calls.cc

```diff
@@ -1099,12 +1099,45 @@ class T_Generic_squareBrackets : public IntrinsicMethod {
         auto it = args.args.begin();
         int i = -1;
         targs.reserve(attachedClass.data(ctx)->typeMembers().size());
+        bool validBounds = true;
         for (auto mem : attachedClass.data(ctx)->typeMembers()) {
             ++i;
-            if (mem.data(ctx)->isFixed()) {
-                targs.emplace_back(mem.data(ctx)->resultType);
+
+            auto memData = mem.data(ctx);
+
+            // TODO: why is mem.data(ctx)->resultType sometimes nullptr?
+            auto *memType = cast_type<LambdaParam>(memData->resultType.get());
+            if (memType != nullptr && memType->isFixed()) {
+                // Fixed args are implicitly applied, and won't consume type
+                // arguments from the list that's supplied.
+                targs.emplace_back(memType->lower);
```

Comment:
why lower?

Reply:
Again, this was the case where the variable is fixed, so the bounds must be the same. I think that this logic can be simplified, and just always select the upper bound.